src/safeposix/syscalls/fs_calls.rs

```diff
@@ -4677,6 +4754,7 @@ impl Cage {
                 // get inodenum for parent of new path
                 let (_, new_par_inodenum) = metawalkandparent(true_newpath.as_path());
                 // check if old and new paths share parent
```

Comment:
why do they  have to share a parent? ive checked the standard, and there seems to be no mention of that

Reply:
basically if they don't share a parent, it would be akin to the move operation, not rename

Reply:
i think it'd be good to explain that in the comments

Reply:
> rename() renames a file, moving it between directories if required. - from https://man7.org/linux/man-pages/man2/rename.2.html

seems it works across directories